Profile model

dapierrot21 opened the pull request:
### What this PR does (required):
- This PR creates a profile model that has fields that represents what is needed from the user.

moffatethan reviewed (changes requested):
Let's change the base of this PR to be the `FESignupPage/LoginPage` so we don't see those files being changed in this PR 🙂. 

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

moffatethan commented on server/models/Profile.js:
```diff
@@ -0,0 +1,55 @@
+const mongoose = require("mongoose");
+
+const profileSchema = new mongoose.Schema({
+    id: {
+        type: String,
+        required: true,
+        unique: true,
+    },
```
We don't need to specify this field - mongodb automatically assigns unique ids to each item in a collection. This can be removed

moffatethan commented on server/models/Profile.js:
```diff
@@ -0,0 +1,55 @@
+        required: true,
+        unique: true,
+    },
+    firstname: {
```
`firstName` let's use camelCase. Change throughout the model please 🙂

moffatethan commented on server/models/Profile.js:
```diff
@@ -0,0 +1,55 @@
+        type: String,
+        required: true,
+    },
+    gender: {
+        type: String,
+        required: true,
+    },
```
Let's do an enum here 

['Male', 'Female', 'Other']

moffatethan commented on server/models/Profile.js:
```diff
@@ -0,0 +1,55 @@
+        type: String,
+        required: true,
+    },
+    birthdate: {
+        type: String,
+        required: true
+    },
```
Let's make this type a `Date` rather than String. It's important we use the proper types for each property otherwise we aren't being very efficient with our data in the back-end.

moffatethan commented on server/models/Profile.js:
```diff
@@ -0,0 +1,55 @@
+        type: String,
+        required: true
+    },
+    email: {
+        type: String,
+        required: true,
+    },
```
The user component has the email address for authentication, we don't need to store this twice.

moffatethan commented on server/models/Profile.js:
```diff
@@ -0,0 +1,55 @@
+        type: String,
+        required: true,
+    },
+    phone: {
+        type: String,
+        required: true,
+    },
```
Would be nice to add some validation to ensure it's a legit phone number coming in via regular expression

moffatethan commented on server/models/Profile.js:
```diff
@@ -0,0 +1,55 @@
+        type: String,
+        required: true,
+    },
+    payment: {
+        type: String,
+        required: true,
+    },
+    hours: {
+        type: Array,
+        required: true,
+    },
+    days: {
+        type: Array,
+        required: true,
+    }
```
We can remove this - hours/days is not going to work out for availability. Will discuss more about this with you when the time comes

moffatethan reviewed (changes requested):
Nice work on this - just some comments to address on the model then this should be good to go. Thanks Dean!

dapierrot21 commented:
I change the base to the correct branch and removed the fields that needed to be remove. Also correct data types and included a reference to the user model. camelCase the fields as well and added regex validations to the phone number field.
